doc/rst/tools/protoc-gen-chpl/protoc-gen-chpl.rst

```diff
@@ -0,0 +1,198 @@
+  in Chapel and provides functionality to serialize messages in wire format.
+
+
+Installation Instructions
+=========================
+
+You should have the protocol buffer compiler installed on your system. For example,
+to install protobuf on an Ubuntu system, you might use ``apt-get install libprotobuf-dev``.
+Also please see the official installation `guide`_.
+
+Once you have the compiler installed, in ``$CHPL_HOME`` run the following:
+
+.. code-block:: sh
+
+   make protoc-gen-chpl
+  
+It builds the protoc-gen-chpl binary so that the command line interface can be used.
+This installs protoc-gen-chpl in the same place as the chapel compiler (``chpl``) so that
+it can be used anywhere in the user's file system.
+
+To remove protobuf suppot, change directory to ``$CHPL_HOME/tools/protoc-gen-chpl`` and run:
+
+.. code-block:: sh
+
+   make clean
+
+
+Defining protocol format
+========================
+
+To make use of the Chapel protobuf implementation you need to start by defining
+a ``.proto`` file. The definitions in a ``.proto`` file contain a message for each
+data structure you want to serialize, then specify a name and a type for each 
+field in the message.
+
+Below is an example of an ``addressbook`` for a person. This section descibes a
+simple ``.proto`` file and the corresponding generated chapel code. For complete
+details on ``.proto`` files see the links at the end of this document.
+
+The Chapel implementation only supports the `proto3`_ syntax of protobuf. The
+``.proto`` file starts with an optional package declaration, which helps to prevent
+naming conflicts between different projects.
+
+.. code-block:: proto
+
+  syntax = "proto3";
+  package addressbook;
+
+In Chapel the generated records will be placed in a module matching the ``package``
+name. If the ``package`` name is not specified the module takes the name of the
+proto file with all non-aplhanumeric characters replaced by an ``underscore``.
+
+Next, you have to define your messages. A message is just an aggregate containing
+a set of typed fields. Many standard simple data types are available as field types,
+including int64, int32, float, double, and string.
+
+.. code-block:: proto
+
+  enum Contact {
+    FAMILY = 0;
+    FRIENDS = 1;
+    WORK = 2;
+  }
+
+  message Person {
+    string name = 1;
+    int32 id = 2;  // Unique ID number for this person.
+    string email = 3;
+
+    Contact contact = 4;
+
+    repeated string number = 5; // Person might have more than one number.
+  }
+  
+In the above example the ``Person`` message contains ``string`` type ``name``
+field with field number ``1``, ``int32`` type ``id`` field with field number
+``2`` as well as ``string`` type ``email`` field with field number ``3``. You
+can also define enum types if you want one of your fields to have one of a
+predefined list of values - here you want to specify that a contact type can be
+one of FAMILY, FRIENDS, or WORK. The ``Contact`` type ``contact`` with field number
```

Comment:
I don't know that you need to call out the type here.  I'd maybe just say "The ``contact`` field" instead